torch/lib/c10d/ProcessGroupGloo.cpp

```diff
@@ -106,6 +109,19 @@ namespace c10d {
 
 namespace {
 
+using steady_clock_time_point =
+    std::chrono::time_point<std::chrono::steady_clock>;
+
+std::chrono::milliseconds getRemainingTime(
+    steady_clock_time_point startTime,
+    const std::chrono::milliseconds& timeout) {
+  auto elapsedTime = std::chrono::steady_clock::now() - startTime;
+  auto elapsedMillis = timeout -
+      std::chrono::duration_cast<std::chrono::milliseconds>(elapsedTime);
+  auto remaining = std::max(elapsedMillis.count(), static_cast<long>(1));
```

Comment:
Not sure why we return 1 here if there is no remaining time? Shouldn't it be 0?

Reply:
Wanted to provide a small buffer of 1 ms so that we call into the recv/send collective, and then it will likely timeout and report the appropriate error. If that seems hacky, we can change this to 0.

Reply:
I don't think we should rely on having a buffer of 1ms to ensure something is called, in different environments the 1ms buffer may or may not work.